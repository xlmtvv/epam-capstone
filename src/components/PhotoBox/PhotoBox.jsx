import React from 'react';
import styles from './PhotoBox.module.scss';

export const PhotoBox = ({ name, title, description, avatar }) => {
  return (
    <div className={styles.photobox}>
      <img src={avatar} alt={`${name}'s avatar`} className={styles.avatar} />
      <div className={styles.info}>
        <h2 className={styles.name}>{name}</h2>
        <h3 className={styles.title}>{title}</h3>
        <p className={styles.description}>{description}</p>
      </div>
    </div>
  );
};
