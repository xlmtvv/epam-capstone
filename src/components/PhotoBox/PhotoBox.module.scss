@use "../../_variables.scss" as *;

.photobox {
  justify-content: center;
  align-items: center;
  text-align: center;
  z-index: 1;
  color: white;
  font-family: $font_family;
}

.avatar {
  width: 163px;
  height: 163px;
  object-fit: cover; 
  border-radius: 50%; 
}

.info {
  flex: 1; 
  max-width: 50vw;
  overflow: hidden;
}

.name {
  font-size: 1.5rem;
  margin-bottom: 5px;
  font-weight: 700;
}

.title {
  font-size: 1.2rem;
  margin-bottom: 10px;
  font-weight: 500;
}
.description {
  font-weight: 300;
  font-size: 0.8em;
}

.info {
  .name {
    color: white;
  }
}

@media (max-width: 400px) {
  .avatar {
    width: 100px;
    height: 100px;
  }

  .name {
    font-size: 35px;
  }

  .title {
    font-size: 20px;
  }
  .description {
    font-size: 15px;
  }
}
