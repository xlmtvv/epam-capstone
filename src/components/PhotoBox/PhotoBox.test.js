import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { PhotoBox } from './PhotoBox';

describe('PhotoBox Component', () => {
  const mockProps = {
    name: 'John Doe',
    title: 'Software Engineer',
    description: 'Experienced in React and Node.js',
    avatar: 'https://example.com/avatar.jpg',
  };

  test('renders without crashing', () => {
    render(<PhotoBox {...mockProps} />);
    expect(screen.getByText(mockProps.name)).toBeInTheDocument();
  });

  test('displays correct props', () => {
    render(<PhotoBox {...mockProps} />);
    expect(screen.getByText(mockProps.name)).toBeInTheDocument();
    expect(screen.getByText(mockProps.title)).toBeInTheDocument();
    expect(screen.getByText(mockProps.description)).toBeInTheDocument();
    expect(screen.getByAltText(`${mockProps.name}'s avatar`)).toHaveAttribute('src', mockProps.avatar);
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<PhotoBox {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});