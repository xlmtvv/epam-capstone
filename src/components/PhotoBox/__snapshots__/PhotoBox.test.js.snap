// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoBox Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="photobox"
  >
    <img
      alt="John Doe's avatar"
      class="avatar"
      src="https://example.com/avatar.jpg"
    />
    <div
      class="info"
    >
      <h2
        class="name"
      >
        John Doe
      </h2>
      <h3
        class="title"
      >
        Software Engineer
      </h3>
      <p
        class="description"
      >
        Experienced in React and Node.js
      </p>
    </div>
  </div>
</DocumentFragment>
`;
